File: TP2_6216948_PW/TP2_6216948_PWS/Models/League.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace TP2_6216948_PWS.Models
{
    public class League
    {
        [Key]
        public int Id { get; set; }

        public string Name { get; set; }

        public string Logo { get; set; }

        //prop nav
        [JsonIgnore]
        public virtual List<Team>? Teams { get; set; }
        [JsonIgnore]
        public virtual List<Saison>? Saisons { get; set; }


    }
}

File: TP2_6216948_Angular/src/Models/League.ts
import { Saison } from "./Saison";
import { Team } from "./Team";

export class League
{
    constructor(public id:number,
                public name:string,
                public logo:string,

                public team:Team[]=[],
                public saisons:Saison[]=[],
                ){

    }
}